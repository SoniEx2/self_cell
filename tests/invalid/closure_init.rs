use once_self_cell::sync::OnceSelfCell;

fn main() {
    let mut s = Ok(".".to_owned());

    let c: OnceSelfCell<(), &String> = OnceSelfCell::new((), |_| s.as_ref().unwrap());
}
