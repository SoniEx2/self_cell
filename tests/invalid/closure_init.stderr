error[E0308]: mismatched types
 --> $DIR/closure_init.rs:6:62
  |
6 |     let c: OnceSelfCell<(), &String> = OnceSelfCell::new((), |_| s.as_ref().unwrap());
  |                                                              ^^^^^^^^^^^^^^^^^^^^^^^ expected fn pointer, found closure
  |
  = note: expected fn pointer `fn(&()) -> &String`
                found closure `[closure@$DIR/tests/invalid/closure_init.rs:6:62: 6:85]`
